servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyChannelPublisher.java

```diff
@@ -166,50 +166,42 @@ private void requestN(long n, SubscriptionImpl forSubscription) {
 
     private boolean processPending(SubscriptionImpl target) {
         // Should always be called from eventloop. (assert done before calling)
-        if (inProcessPending || pending == null) {
-            // Guard against re-entrance.
-            return false;
-        }
-
-        inProcessPending = true;
-        try {
-            while (requestCount > 0) {
-                Object p = pending.poll();
-                if (p == null) {
-                    break;
-                }
-                if (p instanceof TerminalNotification) {
-                    Throwable throwable = ((TerminalNotification) p).cause();
-                    assert throwable != null : "onComplete notification can not be enqueued.";
-                    sendErrorToTarget(target, throwable);
-                    return true;
-                }
-                if (emit(target, p)) {
-                    if (subscription == target || subscription == null) {
-                        // stop draining the pending events if the current Subscription is still the same for which we
-                        // started draining, continue emitting the remaining data if there is a new Subscriber
+        if (!inProcessPending && pending != null && !pending.isEmpty()) {
+            inProcessPending = true;
+            try {
+                while (requestCount > 0) {
+                    Object p = pending.poll();
+                    if (p == null) {
+                        break;
+                    }
+                    if (p instanceof TerminalNotification) {
+                        sendErrorToTarget(target, (TerminalNotification) p);
```

Comment:
the code in this method wasn't meaningfully changed, just refactored to share code and remove the initial return branch.